### PR TITLE
Remove auto-incremented fields from "create implicit actions"

### PR DESCRIPTION
**Pull request by qnkhuat**
Step 2 of https://github.com/metabase/metabase/issues/28946

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/28985)

### CODE REVIEW

File: src/metabase/models/action.clj [hunk 1]
```diff
@@ -245,14 +246,18 @@
...
                                 :always
                                 (map (fn [param] (merge param (get saved-params (:id param)))))
 
-                                (= "row/delete" (:kind action))
+                                (= "row/delete" action-kind)
                                 (filter ::pk?)
 
-                                (contains? #{"row/update" "row/delete"} (:kind action))
+                                (= "row/create" action-kind)
+                                (remove :is-auto-increment)
```
Comment (calherries) [inline src/metabase/models/action.clj:258]: We also need to remove non-required PKs from this too, right? Are you going to do that in another PR?

File: test/metabase/models/action_test.clj [hunk 2]
```diff
@@ -134,3 +134,17 @@
                Exception
                #"Actions must be made with models, not cards"
                (t2/update! Action action-id {:archived false}))))))))
+
+(deftest exclude-non-required-pks-for-create-implicit-actions-test
```
Comment (calherries) [inline test/metabase/models/action_test.clj:138]: The name of this test is too broad. It's only testing auto increment fields

File: src/metabase/models/action.clj (hunk 1 above)
Comment (qnkhuat) [reply src/metabase/models/action.clj:258]: I'm not sure I follow, how is PK nullable?

anw, product specifically asks for autoincrements only ([thread](https://metaboat.slack.com/archives/C04FG88HL95/p1678098385553369?thread_ts=1677593986.943879&cid=C04FG88HL95))

Comment (calherries) [reply src/metabase/models/action.clj:258]: PKs can't be nullable but as we discussed, PKs can be non-required if there's a default.

Bruno was asking specifically about autoincrements, but you were also right [here](https://metaboat.slack.com/archives/C04FG88HL95/p1678089679191799?thread_ts=1677593986.943879&cid=C04FG88HL95). non-required PKs like `id uuid PRIMARY KEY DEFAULT uuid_generate_v4()` should be excluded too.

**Review by calherries — APPROVED** [review]

File: test/metabase/models/action_test.clj (hunk 2 above)
Comment (qnkhuat) [reply test/metabase/models/action_test.clj]: updated

File: src/metabase/models/action.clj (hunk 1 above)
Comment (qnkhuat) [reply src/metabase/models/action.clj:258]: let's wait for Bruno on that, if we want to do that then we will need to track default as well.

I'll resolve this comment for now because I can't merge without resolving it :(

Comment (qnkhuat) [reply src/metabase/models/action.clj:258]: Ah sorry I misunderstood that. yea that makes sense to me too. I'll do that in a follow up
